src/ray/gcs/gcs_client/gcs_client.h

```diff
@@ -241,6 +241,19 @@ class RAY_EXPORT PythonGcsClient {
   Status GetClusterStatus(int64_t timeout_ms, std::string &serialized_reply);
 
  private:
+  inline grpc::ClientContext &&PrepareContext(int64_t timeout_ms) {
+    grpc::ClientContext context;
+    if (timeout_ms != -1) {
+      context.set_deadline(std::chrono::system_clock::now() +
+                           std::chrono::milliseconds(timeout_ms));
+    }
+    if (!cluster_id_.IsNil()) {
+      context.AddMetadata(kClusterIdKey, cluster_id_.Hex());
+    }
+    return std::move(context);
+  }
```

Comment:
```suggestion
 grpc::ClientContext PrepareContext(int64_t timeout_ms) {
    grpc::ClientContext context;
    if (timeout_ms != -1) {
      context.set_deadline(std::chrono::system_clock::now() +
                           std::chrono::milliseconds(timeout_ms));
    }
    if (!cluster_id_.IsNil()) {
      context.AddMetadata(kClusterIdKey, cluster_id_.Hex());
    }
    return context;
  }
```

Reply:
Compiler complains because there's no copy constructor for ClientContext. Moved the ClientContext out of the helper the way it was before.

Reply:
hmmm, interesting... because you actually can return a unique_ptr directly.